book/src/qobject/qobject_struct.md

```diff
@@ -5,43 +5,157 @@ SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
 
 ```rust,ignore,noplayground
-{{#include ../../../examples/qml_features/rust/src/struct_properties.rs:book_macro_code}}
+{{#include ../../../examples/qml_features/rust/src/properties.rs:book_properties_default}}
+```
+
+[Full Example](https://github.com/KDAB/cxx-qt/blob/main/examples/qml_features/rust/src/properties.rs)
+
+### `cxx_type`
+
+You can change the C++ type that your property uses by adding the `cxx_type` attribute to the `#[qproperty]` macro.
+
+This is especially useful if your property type is an opaque C++ type and can't be owned directly by Rust.
+Then it might be necessary to wrap it in a [cxx::UniquePtr](https://docs.rs/cxx/latest/cxx/struct.UniquePtr.html).
+
+However, you may still want your `Q_PROPERTY` to use the opaque type directly.
+This can easily be achieved by using `cxx_type`.
+
+``` rust,ignore,noplayground
+#[cxx_qt::qobject]
+struct MyStruct {
+  #[qproperty(cxx_type="OpaqueExampleType")]
+  my_property: cxx::UniquePtr<OpaqueExampleType>
+}
+```
+In this case, CXX-Qt will automatically convert any references to the `std::unique_ptr<OpaqueExampleType>` to references to `OpaqueExampleType`.
+
+For details, see the [page on type conversions](../concepts/type-conversions.md).
+
+## Invokables
+
+Invokables are functions that are registered with the Qt meta-object system using [`Q_INVOKABLE`](https://doc.qt.io/qt-6/qobject.html#Q_INVOKABLE).
+This allows them to be called from QML and JavaScript (running in a [QJSEngine](https://doc.qt.io/qt-6/qjsengine.html)), in addition to C++ and Rust.
+
+CXX-Qt allows you to define invokables using Rust code.
+This way you can easily add a Rust-powered backend to your QML frontend.
+
+Invokables, by definition, must be defined on a C++ class however.
+This is where the QObject subclass generated by `#[cxx_qt::qobject]` comes into play.
+For details on this, see the [`qobject::T` page](./generated-qobject.md).
+
+The important part for invokables is that they need to be implemented on the `qobject::T`, not `T`.
+Therefore they have access to both C++ and Rust methods. CXX-Qt adds wrapper code around your invokables to automatically convert between the [C++ and Rust types](../concepts/types.md).
+
+To mark a method as invokable, simply add the `#[qinvokable]` attribute to the Rust method. This tells CXX-Qt to expose the method on the generated C++ class.
+`Q_INVOKABLE` will be added to the C++ definition of the method, allowing QML to call the invokable.
+
+``` rust,ignore,noplayground
+{{#include ../../../examples/qml_features/rust/src/invokables.rs:book_impl_qobject}}
 ```
 
-If you want to provide default values for your QObject, then instead of deriving implement the `Default` trait for the struct.
+Note that an Invokable may only use `self: Pin<&mut Self>` or `&self` as self types.
+It is not possible to have a `self`, or `&mut self` invokable, as that may move the QObject in memory, which C++ can't handle.
```

Comment:
```suggestion
It is not possible to have a `self`, or `&mut self` invokable, as that may move the QObject in memory, which would invalidate C++ pointers and references to the QObject.
```

"can't handle" doesn't really explain the issue

Reply:
done